packages/back-end/src/controllers/stripe.ts

```diff
@@ -261,3 +270,118 @@ export async function postWebhook(req: Request, res: Response) {
 
   res.status(200).send("Ok");
 }
+
+export async function postSetupIntent(
+  req: AuthRequest<{ subscriptionId: string }>,
+  res: Response
+) {
+  const { subscriptionId } = req.body;
+  try {
+    const customer = await getStripeCustomerId(subscriptionId);
+    const setupIntent = await stripe.setupIntents.create({
+      customer,
+      payment_method_configuration: STRIPE_PAYMENT_METHOD_CONFIG_ID,
+      automatic_payment_methods: {
+        enabled: true,
+      },
+    });
+    return res.status(200).json({ clientSecret: setupIntent.client_secret });
+  } catch (e) {
+    throw new Error(e.message);
+  }
```

Comment:
We will want to move all of the stripe logic to the license server.  We can keep the API routes here, but they will just call out to the license server instead of stripe directly.  This will allow it to function for self-hosted users as well as cloud.